src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -0,0 +1,299 @@
+                null,
+                secretIndex
+        );
+        if (isSecretPresent) {
+            log.debug("App developer secret found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+            return ResponseEntity.noContent().build();
+        }
+
+        log.debug("App developer secret not found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+        return ResponseEntity
+                .status(HttpStatus.NOT_FOUND)
+                .body(createErrorPayload("Secret not found"));
+    }
+
+    @PostMapping("/{appAddress}/requesters/secrets")
+    public ResponseEntity<Map<String, String>> setMaxRequesterSecretCountForApp(
+            @RequestHeader("Authorization") String authorization,
+            @PathVariable String appAddress,
+            @RequestBody int secretCount) {
+        String challenge = authorizationService
+                .getChallengeForSetRequesterAppRuntimeSecretCount(
+                        appAddress,
+                        secretCount
+                );
+
+        if (!authorizationService.isSignedByOwner(challenge, authorization, appAddress)) {
+            log.error("Unauthorized to setRequesterSecretCountForApp" +
+                            " [appAddress: {}, expectedChallenge: {}]",
+                    appAddress, challenge);
+            return ResponseEntity
+                    .status(HttpStatus.UNAUTHORIZED)
+                    .body(invalidAuthorizationPayload);
+        }
+
+        final boolean isCountAlreadyPresent = teeTaskRuntimeSecretCountService
+                .isMaxAppRuntimeSecretCountPresent(
+                        appAddress,
+                        SecretOwnerRole.REQUESTER
+                );
+        if (isCountAlreadyPresent) {
+            log.debug("Can't add app requester app secret count as it already exist"
+            + " [appAddress:{}]", appAddress);
+            return ResponseEntity
+                    .status(HttpStatus.CONFLICT)
+                    .body(createErrorPayload("Secret count already exists"));
+        }
+
+        final boolean hasBeenInserted = teeTaskRuntimeSecretCountService.setMaxAppRuntimeSecretCount(
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                secretCount
+        );
+
+        if (!hasBeenInserted) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(createErrorPayload(
+                            "Secret count should be positive. " +
+                            "Can't accept value " + secretCount
+                    ));
+        }
+
+        return ResponseEntity.noContent().build();
+    }
+    // endregion
+
+    // region App requester endpoint
+    @PostMapping("/requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<Map<String, String>> addRequesterAppRuntimeSecret(@RequestHeader("Authorization") String authorization,
```

Comment:
We don't. That's a good idea to avoid some spam. Thanks!

Reply:
Added in https://github.com/iExecBlockchainComputing/iexec-sms/pull/69/commits/f771c4f4095de47bb4420fafbc0d3412a23d9e16

Reply:
Thanks!

Reply:
We also need to check that the value of `appContract.getContractAddress()` is not `0x0...0`, otherwise we could set secrets for the address `0x0...0`.

Reply:
Sure, thanks!